src/bika/lims/content/analysisservice.py

```diff
@@ -18,620 +18,332 @@
 # Copyright 2018-2020 by it's authors.
 # Some rights reserved, see README and LICENSE.
 
+import itertools
+
 from AccessControl import ClassSecurityInfo
-from bika.lims import PMF
 from bika.lims import api
 from bika.lims import bikaMessageFactory as _
 from bika.lims.browser.fields import InterimFieldsField
+from bika.lims.browser.fields import PartitionSetupField
 from bika.lims.browser.fields import UIDReferenceField
+from bika.lims.browser.fields.partitionsetupfield import getContainers
 from bika.lims.browser.widgets.partitionsetupwidget import PartitionSetupWidget
 from bika.lims.browser.widgets.recordswidget import RecordsWidget
 from bika.lims.browser.widgets.referencewidget import ReferenceWidget
+from bika.lims.catalog import SETUP_CATALOG
 from bika.lims.config import PROJECTNAME
 from bika.lims.content.abstractbaseanalysis import AbstractBaseAnalysis
 from bika.lims.content.abstractbaseanalysis import schema
 from bika.lims.interfaces import IAnalysisService
 from bika.lims.interfaces import IDeactivable
-from bika.lims.utils import to_utf8 as _c
-from magnitude import mg
+from Products.Archetypes.atapi import PicklistWidget
 from Products.Archetypes.public import BooleanField
 from Products.Archetypes.public import BooleanWidget
 from Products.Archetypes.public import DisplayList
-from Products.Archetypes.public import MultiSelectionWidget
 from Products.Archetypes.public import Schema
 from Products.Archetypes.public import SelectionWidget
 from Products.Archetypes.public import registerType
-from Products.Archetypes.Registry import registerField
 from Products.CMFCore.utils import getToolByName
-from senaite.core.browser.fields.records import RecordsField
 from zope.interface import implements
 
+Methods = UIDReferenceField(
+    "Methods",
+    schemata="Method",
+    required=0,
+    multiValued=1,
+    vocabulary="_methods_vocabulary",
+    allowed_types=("Method", ),
+    accessor="getRawMethods",
+    widget=PicklistWidget(
+        label=_("Methods"),
+        description=_(
+            "Available methods to perform the test depending if they "
+            "allow or deny the manual entry of results."),
```

Comment:
Replace by "Available methods to perform the test"

Reply:
Done